### DISCUSSION TITLE
Can i compile and link to this library?

### DISCUSSION THREAD

**Discussion by enesaltinkaya** [nlohmann/json#3840] [General]
When i include json.hpp, if a .cpp file changes using this header, compile times slow down.

Is it possible to compile this library as .lib or .so then include to my project, so that it wont get compiled over and over again?

Thanks.

**Comment by nlohmann** [comment]
No, the library is header only due to heavy use of templates. There is now way to compile is as a library.

**Comment by enesaltinkaya** [comment]
Alright.

I managed to abstract json.hpp's functionality, so that it is only included in 1 cpp file.
This way it is not getting compiled all the time.

Thanks for the awesome library.

**Reply by nlohmann** [reply]
Note there is also a forward declaration header, `json_fwd.hpp` which may help to reduce dependencies to the big header.
